cloudigrade/account/tests/test_util.py

```diff
@@ -0,0 +1,145 @@
+"""Collection of tests for utils in the account app."""
+import random
+import uuid
+from unittest.mock import patch
+
+from django.test import TestCase
+
+from account import AWS_PROVIDER_STRING, util
+from account.models import AwsAccount, AwsMachineImage
+from util import aws
+from util.tests import helper as util_helper
+
+
+def mock_publish(body, exchange, routing_key, declare):
+    """Mock queue publish function."""
+    return True
+
+
+class AccountUtilTest(TestCase):
+    """Account util test cases."""
+
+    def test_create_new_machine_images(self):
+        """Test that new machine images are saved to the DB."""
+        aws_account_id = util_helper.generate_dummy_aws_account_id()
+        arn = util_helper.generate_dummy_arn(aws_account_id)
+        account = AwsAccount(account_arn=arn, aws_account_id=aws_account_id)
+        account.save()
+
+        region = f'region-{uuid.uuid4()}'
+        running_instances = {
+            region: [
+                util_helper.generate_dummy_describe_instance(
+                    state=aws.InstanceState.running
+                )
+            ]
+        }
+        ami_id = running_instances[region][0]['ImageId']
+
+        result = util.create_new_machine_images(account, running_instances)
+
+        amis = AwsMachineImage.objects.filter(account=account).all()
+
+        self.assertEqual(result, [ami_id])
+        for ami in amis:
+            self.assertEqual(ami.ec2_ami_id, ami_id)
+
+    def test_create_new_machine_images_with_windows_image(self):
+        """Test that new windows machine images are marked appropriately."""
+        aws_account_id = util_helper.generate_dummy_aws_account_id()
+        arn = util_helper.generate_dummy_arn(aws_account_id)
+        account = AwsAccount(account_arn=arn, aws_account_id=aws_account_id)
+        account.save()
+
+        region = f'region-{uuid.uuid4()}'
```

Comment:
I've been meaning to clean up some older tests that do the same thing, but to get us closer to somewhat more realistic mock data, I'd like to take all our goofy UUID-based fake regions and swap them out with `random.choice(helper.SOME_AWS_REGIONS)`.

If you're feeling ambitious, you could do that here and in your other new tests… or just punt this for later. I have no strong opinion at the moment.

Reply:
Oh, I see you're actually using `random.choice` in some of the later tests! In that case, yeah, let's clean this up for consistency.

Reply:
Oops. Will do.